spec/lib/query/query_payments_spec.rb

```diff
@@ -294,6 +295,27 @@ def generate_donation(h)
         end
       end
 
+      context 'filtering by payment id' do
+        let(:input) {{
+          amount: 100,
+          nonprofit_id: nonprofit.id,
+          supporter_id: supporter.id,
+          token: source_tokens[4].token,
+          date: (Time.now - 1.day).to_s,
+          comment: 'donation comment',
+          dedication: 'dedication',
+          designation: 'designation'
+        }}
+
+        it 'returns one result' do
+          InsertDonation.with_stripe(input)
+          donation_result_tomorrow
+          donation_result_yesterday
+
+          result = QueryPayments::full_search(nonprofit.id, { search: Payment.last.id })
+          expect(result[:data].count).to eq 1
+        end
+      end
```

Comment:
Could you add a test that makes sure it still finds a payments if an integer is in the notes but isn't the payment ID itself?

Reply:
I added one test for this :smiley: